src/main/kotlin/org/malachite/estella/services/TaskStageService.kt

```diff
@@ -116,25 +118,57 @@ class TaskStageService(
         taskStageRepository.save(taskStage.copy(tasksResult = newTaskResults))
     }
     private fun copyTaskResult(foundResult: TaskResult?, resultToAdd: TaskService.ResultToAdd) =
-        foundResult?.copy(results = resultToAdd.results, code = resultToAdd.code, startTime = foundResult.startTime, endTime = resultToAdd.time, task = resultToAdd.task, taskStage = resultToAdd.taskStage)
-    private fun createNewTaskResult(resultToAdd: TaskService.ResultToAdd) =
-        TaskResult(null, resultToAdd.results, resultToAdd.code, resultToAdd.time, null, resultToAdd.task, resultToAdd.taskStage)
+        foundResult?.copy(
+                results = resultToAdd.results,
+                code = resultToAdd.code,
+                startTime = foundResult.startTime,
+                endTime = resultToAdd.time,
+                task = resultToAdd.task,
+                taskStage = resultToAdd.taskStage
+        )
+
+    private fun createNewTaskResult(resultToAdd: TaskService.ResultToAdd): TaskResult {
+        if (isFirstSolvedTask(resultToAdd))
+            sendDevNotification(resultToAdd)
+
+        return TaskResult(null, resultToAdd.results, resultToAdd.code, resultToAdd.time, null, resultToAdd.task, resultToAdd.taskStage)
```

Comment:
Yyy tak w sumie czemu ta funkcja nie zapisuje od razu do bazy TaskResult'a?

Reply:
popatrz sobie funkcje poziom wyżej - to jest jeden z dwóch scenariuszy który zwraca obiekt TaskResult, i zapisywany jest później.  imo tak jest git

Reply:
Znaczy jak na moje to jest kiepskie bo jeśli ktoś by chciał reużyć tej funkcji to może wysłać maila a nie zapisze w bazie. Więc imo albo bym zrobił funkcję która tylko tworzy obiekt (bez side-effectów) albo taką co obsługuje wszystkie side-effecty. Bo to jest dosyć mylące, że stworzy ci obiekt nie zapisze w bazie, ale wyśle maila który zależy od tego czy masz coś zapisane w bazie XD

Reply:
ok racja XD z tym że teraz jak będziemy mieli nowy endpoint na startowanie zadania, to ta funkcja w ogóle nie będzie potrzebna więc w sumie nie ma znaczenia xD